9armbot-2.0/services/twitch.ts

```diff
@@ -89,9 +101,17 @@ export async function twitchService() {
 
   client.on('message', async (channel, tags, message, self) => {
     if (self) return
-    if (message[0] !== '!') return
 
     const username = tags!.username!
+
+    // Ensure new players are created on message
+    if (username && !existingPlayerNames.has(username.toLowerCase())) {
```

Comment:
On user message : check if username is already in the set, create player if not.

This will cover when bot is restarted in live session, or join after the live session started